datumaro/plugins/data_formats/datumaro/exporter.py

```diff
@@ -38,7 +38,7 @@
 
 
 class _SubsetWriter:
-    def __init__(self, context):
+    def __init__(self, context: IDataset, ann_file: str):
```

Comment:
There must be some reasons to get `ann_file` during initialization. Could you clarify this?

Reply:
To reuse the codes for Binary format with different extensions (.json -> .datumaro)?

Reply:
Yes, Because `DatumaroBinaryFormat` requires having `ann_file` at the initialization, so I refactored it.